lib/elixir/src/elixir_tokenizer.erl

```diff
@@ -194,7 +194,7 @@ tokenize([$~, S, H, H, H | T] = Original, Line, Column, Scope, Tokens) when ?is_
   case extract_heredoc_with_interpolation(Line, Column, Scope, ?is_downcase(S), T, H) of
     {ok, NewLine, NewColumn, Parts, Rest} ->
       {Final, Modifiers} = collect_modifiers(Rest, []),
-      tokenize(Final, NewLine, NewColumn, Scope, [{sigil, {Line, Column, NewColumn}, S, Parts, Modifiers} | Tokens]);
+      tokenize(Final, NewLine, NewColumn, Scope, [{sigil, {Line, Column, NewColumn}, S, Parts, Modifiers, [H, H, H]} | Tokens]);
```

Comment:
The terminator sometimes is a list and sometimes it is a single integer. We need to pick one or the other. The advantage of being a list is that we can express heredocs and non-heredocs in the same attribute. Otherwise we will need another metadata field, like `[heredoc: true]`. Thoughts?